great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1360,20 +1363,27 @@ def get_batch_data_and_markers(
         create_temp_table: bool = batch_spec.get(
             "create_temp_table", self._create_temp_table
         )
+        # this is where splitter components are added to the selectable
+        selectable: sqlalchemy.Selectable = self._build_selectable_from_batch_spec(
+            batch_spec=batch_spec
+        )
+        # NOTE: what's being checked here is the presence of a `query` attribute, we could check this directly
+        # instead of doing an instance check
         if isinstance(batch_spec, RuntimeQueryBatchSpec):
             # query != None is already checked when RuntimeQueryBatchSpec is instantiated
-            query: str = batch_spec.query
-
+            # re-compile the query to include any new parameters
+            compiled_query = selectable.compile(
+                dialect=self.engine.dialect,
+                compile_kwargs={"literal_binds": True},
+            )
+            query_str = str(compiled_query)
```

Comment:
@billdirks re-compiling the query

Reply:
Yes, this is ugly but I am ok with this at the moment. I think it would be better to understand and unify these 2 branches but that could be a much bigger change and it sounds like we need to get this fixed ASAP. Could you file a ticket to unify these branches? You can put it in https://go/v1-jira